csharp/src/Ice/InputStream.cs

```diff
@@ -1111,11 +1111,11 @@ internal InputStream(Communicator communicator, Encoding encoding, ArraySegment<
             (int Size, Encoding Encoding) result = ReadEncapsulationHeader(Encoding, _buffer.Slice(_pos));
             if (OldEncoding)
             {
-                _pos += 6; // 4 (size length) + 2 (encoding length)
+                _pos += 6; // 4 (size length) + 2 (encoding bytes)
```

Comment:
Maybe remove length here too
```suggestion
                _pos += 6; // 4 (size bytes) + 2 (encoding bytes)
```

Reply:
Fixed.